runners/direct-java/src/main/java/org/apache/beam/runners/direct/DirectRunner.java

```diff
@@ -282,21 +283,32 @@ public DirectPipelineResult run(Pipeline pipeline) {
     Collection<ModelEnforcementFactory> parDoEnforcements = createParDoEnforcements(options);
     enforcements.put(ParDo.Bound.class, parDoEnforcements);
     enforcements.put(ParDo.BoundMulti.class, parDoEnforcements);
+    if (options.isEnforceEncodability()) {
```

Comment:
This seems fine, although going forward (if we add more enforcements) it might be easier to have a list of enabled enforcements, and a list of what transforms they apply to, rather than having to build up a single mapping of all the enforcements for each class in one place (specifically, it would be nice if EncodabilityEnforcementFactory new which option it was gated on and which transforms it applied to).